### ISSUE TITLE
Zone monkey patching of Promise.all breaks generator functions

### ISSUE THREAD

**Issue by eamodio** [Microsoft/ApplicationInsights-node.js#267]
I have a generator function:

```ts
    *resolveRules(rules: IRule[], fileName: string, document: TextDocument, rootPath: string | undefined): Iterable<Promise<Uri[]>> {
        for (const rule of rules) {
            yield *rule.provideRelated(fileName, document, rootPath);
        }
    }
```
See [here](https://github.com/eamodio/vscode-find-related/blob/master/src/rulesProvider.ts#L56)

That I pass to `Promise.all`
```ts
const files = await Promise.all<Uri[]>(this.rulesProvider.resolveRules(rules, fileName, document, workspace.rootPath));
```
See [here](https://github.com/eamodio/vscode-find-related/blob/master/src/commands/showRelated.ts#L59)

And because of the Zone monkey patching the generator function never gets called

**Comment by AlexBulankou** [comment]
@OsvaldoRosado, can this one be closed with the referenced changes?

**Comment by OsvaldoRosado** [comment]
I don't think so. We need to make sure we don't break native promises in generators by default

**Comment by OsvaldoRosado** [comment]
This is now fixed in 1.1.0-beta.0. Verified with the following snippet:

```
var appInsights = require('applicationinsights');
appInsights.setup('ikey').start();

var generator = function*() {
    yield Promise.resolve('1');
    yield Promise.resolve('2');
    yield Promise.resolve('3');
    return;
};

Promise.all(generator()).then(val=>{
    console.log(val);
})
```

This snippet prints `[]` in 1.0.8 and correctly returns `[ '1', '2', '3' ]` in 1.1.0-beta.0. 

Feel free to re-open if you're still running into this problem!
